internal/connector/internal/config/connectors.go

```diff
@@ -43,6 +44,12 @@ type ConnectorCatalogEntry struct {
 	ConnectorType public.ConnectorType              `json:"connector_type"`
 }
 
+type ConnectorMetadata struct {
+	ConnectorTypeId string   `json:"id" yaml:"id"`
+	FeaturedRank    int32    `json:"featured-rank" yaml:"featured-rank"`
+	Labels          []string `json:"labels" yaml:"labels"`
```

Comment:
There is no need to use StructTag for `Labels` since the default conversion (for both json and yaml packages) is to `labels`.

Also it is the common approach to use underscore instead of minus, any reason why minus for FeatureRank here makes more sense?